inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -1802,3 +1807,42 @@ void MKLDNNGraphOptimizer::MergeTransposeAndReorder(MKLDNNGraph &graph) {
         }
     }
 }
+
+void MKLDNNGraphOptimizer::reshapeRnnSeq(MKLDNNGraph &graph) {
+    auto& graphNodes = graph.GetNodes();
+
+    auto isSutableParentNode = [](MKLDNNNodePtr node) {
+        if (node->type != RNNSeq)
+            return false;
+        auto rnnNode = std::dynamic_pointer_cast<MKLDNNRNN>(node);
+        return rnnNode && !rnnNode->hasNativeOrder() && node->outDims[0].ndims() == 4 && node->outDims[0][1] == 1;
+    };
+
+    auto parent = graphNodes.begin();
+    while (parent != graphNodes.end()) {
+        auto parentNode = *parent;
+        if (!isSutableParentNode(parentNode)) {
+            parent++;
+            continue;
+        }
+
+        auto childrenEdges = parentNode->getChildEdgesAtPort(0);
+        auto newRnnOutDims = parentNode->outDims[0].ToSizeVector();
+        newRnnOutDims.erase(newRnnOutDims.begin() + 1);
+        parentNode->outDims[0] = MKLDNNDims{newRnnOutDims};
+
+        for (size_t i = 0; i < childrenEdges.size(); i++) {
+            auto edge = childrenEdges[i];
+            auto childNode = edge->getChild();
+
+            const MKLDNNNodePtr newReshape = std::make_shared<MKLDNNReshapeNode>(parentNode,
+                    parentNode->getName() + "_abc_a1bc_" + std::to_string(i), graph.getEngine(), graph.weightsCache);
+            newReshape->outDims.push_back(childNode->inDims[edge->getOutputNum()]);
```

Comment:
I believe better to pass inDims, outDims and precision directly in ctor

Reply:
FIxed